dashboard/test/models/pd/application/teacher_application_test.rb

```diff
@@ -839,20 +855,21 @@ class TeacherApplicationTest < ActiveSupport::TestCase
     end
 
     test 'scholarship criteria uses regional partner set fields when specified' do
-      regional_partner = build :regional_partner,
+      regional_partner = create :regional_partner,
         frl_guardrail_percent: REGIONAL_PARTNER_DEFAULT_GUARDRAILS[:frl_not_rural] + 2,
         urg_guardrail_percent: REGIONAL_PARTNER_DEFAULT_GUARDRAILS[:urg] - 2
 
       # Does not meet Free and Reduced Lunch criteria but does meet Underrepresented Group criteria
       principal_options = Pd::Application::PrincipalApprovalApplication.options
-      application_hash = build :pd_teacher_application_hash,
+      application_hash = build TEACHER_APPLICATION_HASH_FACTORY,
```

Comment:
Good question! It's super inconsistently applied –– I'd be down to add a tech debt item to clean this up because I don't think it's necessary.

I think historically, the factory names were by year:
https://github.com/code-dot-org/code-dot-org/blob/a8c72bea688509eccaa785f24569b2cdd4477ad8/dashboard/app/models/pd/application/active_application_models.rb#L16-L25

I think this is because sometimes an entirely new model was made for each year, e.g. `Pd::FitWeekend1920Registration` –– so the corresponding factory was `:pd_fit_weekend1920_registration`. I'm wondering if the `TeacherApplication` and/or `PrincipalApprovalApplication` followed the same pattern for a while –– there were different ones for each year?

Regardless of the exact pattern of the model name for Teacher Applications, teacher application *factories* were also by year for a while. It looks like Tim fairly recently went in and removed the year from the factories (https://github.com/code-dot-org/code-dot-org/commit/a8c72bea688509eccaa785f24569b2cdd4477ad8). So I think at this point, it's not necessary –– we have a single factory, and if the contents of the application change / the requirements change from year to year, we just modify the factory accordingly.

Reply:
Ahh, thanks for this, the year explanation makes a lot of sense! Seems like many of those constants (not just factory names) could be removed.